src/query/plan/rule_based_planner.hpp

```diff
@@ -190,19 +194,33 @@ class RuleBasedPlanner {
         uint64_t merge_id = 0;
         uint64_t subquery_id = 0;
 
+        std::unordered_map<std::string, std::shared_ptr<LogicalOperator>> pc_ops;
+
+        if (single_query_part.pattern_comprehension_matchings.size() > 1) {
+          throw utils::NotYetImplemented("Multiple pattern comprehensions.");
+        }
+        for (const auto &matching : single_query_part.pattern_comprehension_matchings) {
+          std::unique_ptr<LogicalOperator> new_input;
+          MatchContext match_ctx{matching.second, *context.symbol_table, context.bound_symbols};
+          new_input = PlanMatching(match_ctx, std::move(new_input));
+          std::vector<NamedExpression *> exprs = {matching.second.result_expr};
+          new_input = std::make_unique<Produce>(std::move(new_input), exprs);
```

Comment:
You can use CTAD for type deduction so no need for NamedExpression* and with temporary it should be forward in init without copy
```suggestion
new_input = std::make_unique<Produce>(std::move(new_input), std::vector{matching.second.result_expr});
```